src/linter.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{AnyStatement, ConnectionTrait, StatementBuilderPlugin};
+
+#[derive(Debug)]
+pub struct QueryLinter<'a, C>
+where
+    C: ConnectionTrait,
+{
+    db: &'a C,
+}
+
+impl<'a, C> QueryLinter<'a, C>
+where
+    C: ConnectionTrait,
+{
+    pub fn new(db: &'a C) -> Self {
+        Self { db }
+    }
+}
+
+impl<C> StatementBuilderPlugin for QueryLinter<'_, C>
+where
+    C: ConnectionTrait,
+{
+    fn run(&self, stmt: &AnyStatement) {
+        match stmt {
+            AnyStatement::Insert(stmt) => InsertQueryLinter::run(stmt),
+            AnyStatement::Select(stmt) => SelectQueryLinter::run(stmt),
+            AnyStatement::Update(stmt) => UpdateQueryLinter::run(stmt),
+            AnyStatement::Delete(stmt) => DeleteQueryLinter::run(stmt),
+            _ => {}
+        }
+    }
+}
```

Comment:
Linter should be defined in a separate crate. But for simplicity and demo purpose. I just place it inside SeaORM.